overlord/state/change_test.go

```diff
@@ -223,3 +161,60 @@ func (cs *changeSuite) TestState(c *C) {
 
 	c.Assert(chg.State(), Equals, st)
 }
+
+func (cs *changeSuite) TestErr(c *C) {
+	st := state.New(nil)
+	st.Lock()
+	defer st.Unlock()
+
+	chg := st.NewChange("install", "...")
+
+	t1 := st.NewTask("download", "Download")
+	t2 := st.NewTask("activate", "Activate")
+
+	chg.AddTask(t1)
+	chg.AddTask(t2)
+
+	c.Assert(chg.Err(), IsNil)
+
+	t1.SetStatus(state.ErrorStatus)
+	c.Assert(chg.Err(), IsNil)
```

Comment:
I tripped over this when reading the test, i.e. I was wondering why t1 in ErrorState is giving chg.Err() == nil and t2 != nil. Maybe a small comment like `// t2 still running` (or not in an end-state yet) or something to give a hint to the reader?